pkg/agent/execbridge.go

```diff
@@ -150,6 +177,13 @@ func (h *execMonitorHandle) Downscale(
 		h.runner.recordResourceChange(current, target, h.runner.global.metrics.monitorApprovedChange)
 	}
 
+	h.runner.status.update(h.runner.global, func(ps podStatus) podStatus {
```

Comment:
You can call this in an `else` for an above `if` and get rid of an `if` inside. 

Same for other status updates.

Reply:
Fixed in a2f0bcf6a60a92aa8c53cd6b52749a76f7bcbe90